### PR TITLE
Support fake-hardware

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,11 +13,12 @@ RUN cp /etc/ironic/ironic.conf /etc/ironic/ironic.conf_orig && \
...
-    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_power_interfaces ipmitool,idrac && \
-    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_management_interfaces ipmitool,idrac && \
-    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_hardware_types ipmi,idrac && \
-    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_vendor_interfaces ipmitool,no-vendor,idrac && \
+    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_boot_interfaces fake,pxe,ipxe && \
+    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_power_interfaces fake,ipmitool,idrac && \
+    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_management_interfaces fake,ipmitool,idrac && \
+    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_hardware_types fake-hardware,ipmi,idrac && \
+    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_vendor_interfaces fake,ipmitool,no-vendor,idrac && \
+    crudini --set /etc/ironic/ironic.conf DEFAULT enabled_deploy_interfaces fake,direct && \
```
Comment (dtantsur) [inline Dockerfile]: nit: it does not matter, but I'd prefer `fake` to go last.

File: runironic.sh [hunk 2]
```diff
@@ -45,6 +45,8 @@ if ! iptables -C INPUT -i $INTERFACE -p tcp -m tcp --dport 5050 -j ACCEPT > /dev
     iptables -I INPUT -i $INTERFACE -p tcp -m tcp --dport 5050 -j ACCEPT
 fi
 
+mkdir -p /shared/html
```
Comment (dtantsur) [inline runironic.sh]: Unrelated?

Comment (stbenjam) [reply runironic.sh]: It doesn’t seem to start without this. Ironic conductor looks for /shared/html/boot.pxe.  On metalkube we create this directory in dev scripts

Comment (bfournie) [reply runironic.sh]: That fix was added to runhttpd.sh in https://github.com/metalkube/metalkube-ironic/commit/7f73fc99a06d7c1e24791dccdb9576d1e801bf80 so it shouldn't be needed here.

Comment (stbenjam) [reply runironic.sh]: I would like to not have to run anything but the ironic container for my tests.  With fake-hardware, is there a reason to require httpd? There's no real nodes that are going to be hitting it

Comment (bfournie) [reply runironic.sh]: OK, fine to leave it here then.  Although it shouldn't be needed /shared/html is defiuned as http_root in ironic.conf.

**Comment by stbenjam** [comment]
Updated
